### PR TITLE
Share manager for managing shared directories with different permissions

### PR DESCRIPTION
**Pull request by zelfroster**
Wrote initial code for Share Manager, Made a SCSS mixin for showing information according to type (concern level).
1692208990_Aug16_23:33:10
1692208970_Aug16_23:32:50
1692208945_Aug16_23:32:25

### CODE REVIEW

**Comment by zelfroster** [comment]
Shared Directories currently looks like this,
image

I have used the table layout and the cells are input elements, So, initially inputs are disabled And, when you click Edit button, then inputs are enabled, so it looks like this
image

But I am thinking about How to approach the last column (Visibility), Should I create another popup for it like in the Qt app? @rottencandy 
1692593993_Aug21_10:29:53

**Comment by zelfroster** [comment]
Share Manager is complete and It works.
- When user first clicks on **Configured Shared Directories** in My Files section.
image
- When the user clicks on `Edit` button on the bottom of the popup, They can edit the entries.
image
- Add New Directory Popup
image
- Manage Visibility Popup
image

This PR is now ready for review @csoler.

**Comment by zelfroster** [comment]
Added Icons for share-flags instead of plain checkbox.
- Normal View
image
- Edit View
image

**Comment by zelfroster** [comment]
Hey @csoler, If you need anything else to be done in this PR, please let me know. If not, then this is ready for review.

**Comment by csoler** [comment]
It really looks good. I just found a few bugs:

- user defined groups appear as "undefined" in the Visibility column, and are not listed when editing groups to add/remove
- when removing a group for a given share, the group is never removed
- when adding a group for a given share, the visibility column does not update immediately. It needs the user to click somewhere before.
- when adding a new directory, it says "successfuly added" but the list does not update, although retroshare is already aware of this new directory (I can see it appear in the qt UI)

All these should be easy to fix!

**Comment by zelfroster** [comment]
> 

Ohh alright, Thanks for pointing these out. I will fix these.

**Comment by zelfroster** [comment]
Hello @csoler, I didn't knew there were other groups as well, I previously had simply copied the RsNodeGroupId from [rspeers.h 190](https://github.com/RetroShare/libretroshare/blob/b46e30caaa204c243126e741696e8f372044715d/src/retroshare/rspeers.h#L190) in libretroshare.

Now, I have implemented that as well. User defined groups (**Test** in the below example) now appear in the Share Manager.
1693795731_Sep04_08:18:51

**Comment by zelfroster** [comment]
I have fixed all the bugs you listed. I hope there's none left.
One more thing, there are no events for the actions in Share Manager, so Webui can't be updated if something is changed in Qt App.

**Comment by zelfroster** [comment]
Now share manager will respond to rs events and update accordingly.

https://github.com/RetroShare/RSNewWebUI/assets/93514460/d1d87a4c-9fa9-4b3b-a97e-f3c0f7273c79

File: webui-src/app/files/files_manager.js [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+                  'td',
+                  {
+                    // since this is not an input element, manually change color
+                    style: { color: isEditDisabled ? '#6D6D6D' : 'black' },
+                    onclick: () =>
+                      !isEditDisabled && widget.popupMessage(m(ManageVisibility, { parentGroups })),
+                  },
+                  parentGroups.length === 0
+                    ? 'All Friend nodes'
+                    : parentGroups.map((groupFlag) => `${futil.RsNodeGroupId[groupFlag]},`)
```
Comment (rottencandy) [inline webui-src/app/files/files_manager.js:230]: nit: you could do a join so `,` at the end isn't added
```suggestion
                    : parentGroups.map((groupFlag) => futil.RsNodeGroupId[groupFlag]).join(', ')
```

File: webui-src/app/files/files_manager.js [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+    });
+  }
+  return {
+    oninit: loadSharedDirectories,
+    view: () => {
+      return m('.widget', [
+        m('.widget__heading', m('h3', 'ShareManager')),
+        m('form.widget__body.share-manager', { onsubmit: setNewSharedDirectories }, [
+          m('blockquote.info', shareManagerInfo),
+          sharedDirArr.length && m(ShareDirTable),
```
Comment (rottencandy) [inline webui-src/app/files/files_manager.js:253]: This check is already happening in the component
```suggestion
          m(ShareDirTable),
```

File: webui-src/app/scss/pages/_files.scss [hunk 3]
```diff
@@ -178,3 +182,115 @@ table.friendsfiles td:nth-child(2) {
...
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  &__table {
+    margin: 1rem 0 auto;
+    thead {
+      font-weight: bold;
+      td:nth-child(3),
+      td:nth-child(4) {
+        text-align: left;
```
Comment (rottencandy) [inline webui-src/app/scss/pages/_files.scss:215]: I feel like we should align:left all headings, the first & second headings look a bit out of place, what do you think?

Comment (zelfroster) [reply webui-src/app/scss/pages/_files.scss:215]: Yeah, but I also did the same in start, but the table cells are input elements and have some padding, so It doesn't look good.
1694231972_Sep09_09:29:32

I can maybe add some padding to the headings as well and then do it. Is that good?
1694232295_Sep09_09:34:55

File: webui-src/app/files/files_manager.js (hunk 2 above)
Comment (zelfroster) [reply webui-src/app/files/files_manager.js]: Sorry for this 😅

File: webui-src/app/files/files_manager.js (hunk 1 above)
Comment (zelfroster) [reply webui-src/app/files/files_manager.js:230]: Yup, that's much better. Thanks : )

File: webui-src/app/scss/pages/_files.scss (hunk 3 above)
Comment (rottencandy) [reply webui-src/app/scss/pages/_files.scss]: yeah that sounds better

**Review by rottencandy — APPROVED** [review]
Looks good, thanks.

**Pull request merged**